components/list/list-item-expand-collapse-mixin.js

```diff
@@ -0,0 +1,161 @@
+import '../button/button-icon.js';
+import '../loading-spinner/loading-spinner.js';
+import { css, html, nothing } from 'lit';
+import { EventSubscriberController } from '../../controllers/subscriber/subscriberControllers.js';
+
+const dragIntervalDelay = 100;
+const dragHoverDropTime = 1000;
+
+export const ListItemExpandCollapseMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether to show the expand collapse toggle
+			 * @type {boolean}
+			 */
+			expandable: { type: Boolean },
+			/**
+			 * Default state for expand collapse toggle - if not set, collapsed will be the default state
+			 * @type {boolean}
+			 */
+			expanded: { type: Boolean, reflect: true },
+			_siblingHasNestedItems: { state: true },
+			_renderExpandCollapseSlot: { type: Boolean, reflect: true, attribute: '_render-expand-collapse-slot' },
+			_showNestedLoadingSpinner: {  state: true }
+		};
+	}
+
+	static get styles() {
+		const styles = [ css`
+			:host {
```

Comment:
The animation should be a progressive enhancement for when the user has no motion preference.
```javascript
@media (prefers-reduced-motion: no-preference) {
	.d2l-list-expand-collapse {
		transition: width var(--d2l-expand-collapse-slot-transition-duration) cubic-bezier(0, 0.7, 0.5, 1);
	}
}
```